pkg/qdsa/Makefile.dep

```diff
@@ -6,5 +6,20 @@ ifneq (,$(filter arch_avr8,$(FEATURES_USED)))
   USEMODULE += qdsa_asm
 endif
 
+ifeq (,$(filter qdsa_impl_%,$(USEMODULE)))
+  ifneq (,$(filter cortex-m23 cortex-m0%,$(CPU_CORE)))
+    USEMODULE += qdsa_impl_arm
+  else ifneq (,$(filter arch_avr8,$(FEATURES_USED)))
+    USEMODULE += qdsa_impl_avr
+  else
+    # on first iteration check for avr8
+    ifeq (,$(filter arch_avr8,$(FEATURES_OPTIONAL)))
```

Comment:
I don't get this. Any provided arch feature should always be used, shouldn't they? They cannot be optional, as you cannot turn on/off an MCUs architecture, right?

Reply:
That would make sense, but that's not the case currently AFAIK. We would need a special handling for `arch_` features, right?

This was an attempt to check if the feature was provided, because we are not allowed to check `FEATURES_PROVIDED` directly, but I just doubled checked, and turns out we can't check `FEATURES_OPTIONAL` either (https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/buildsystem_sanity_check/check.sh).

Something like what you suggest would solve this limitation, as we can still check `FEATURES_USED`, but it should be on the list before the first dependency resolution iteration.

Reply:
This special handling is for the old dependency system in place (but racy if dependency resolution depends on the order how things happen):

https://github.com/RIOT-OS/RIOT/blob/7da50b05a951766ed2c641bf33ef9316d782bb8b/Makefile.dep#L96-L100

I think we should have the same with KConfig (but ideally without the racy flaws of the old system :-))

Reply:
> This special handling is for the old dependency system in place (but racy if dependency resolution depends on the order how things happen):

IMO then this should happen first, before anything. Those features are there before resolving any dependency.

> I think we should have the same with KConfig (but ideally without the racy flaws of the old system :-))

No need, we can check those symbols in Kconfig:

https://github.com/RIOT-OS/RIOT/pull/16929/files#diff-dcf3b7a9b4402beca490254555a63fc04e9927978d8ef40cac56208ea65b6dd5R20-R21

Reply:
With #17224 this check now is simpler (and should not make the static checks angry), thanks!